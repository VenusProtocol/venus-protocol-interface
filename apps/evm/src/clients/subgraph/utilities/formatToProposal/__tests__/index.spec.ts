import fakeAccountAddress from '__mocks__/models/address';
import proposalsResponse from '__mocks__/subgraph/proposals.json';
import BigNumber from 'bignumber.js';
import type { ProposalsQuery } from 'clients/subgraph/gql/generated/governanceBsc';
import { formatToProposal } from '..';

describe('formatToProposal', () => {
  beforeEach(() => {
    vi.useFakeTimers().setSystemTime(new Date(1710401645000));
  });

  it('returns proposal in the correct format', async () => {
    const res = formatToProposal({
      gqlProposal: proposalsResponse.proposals[0] as ProposalsQuery['proposals'][number],
      currentBlockNumber: 41360384,
      proposalMinQuorumVotesMantissa: new BigNumber('1000000000000000000000'),
      proposalExecutionGracePeriodMs: 1000,
      accountAddress: fakeAccountAddress,
      blockTimeMs: 3000,
    });

    expect(res).toMatchSnapshot();
  });
});
