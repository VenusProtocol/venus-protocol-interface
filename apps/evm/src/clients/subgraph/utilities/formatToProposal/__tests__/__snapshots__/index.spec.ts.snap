// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`formatToProposal > returns proposal in the correct format 1`] = `
{
  "abstainVotes": [
    {
      "address": "0x21f62c4c380264231d33dd1a1fe546d013b905f8",
      "proposalId": "353",
      "reason": undefined,
      "support": 2,
      "votesMantissa": "801851714101606493084",
    },
  ],
  "abstainedVotesMantissa": "801851714101606493084",
  "againstVotes": [
    {
      "address": "0x1abe7f510dd4cc2b017ef78f4757b83ae334862d",
      "proposalId": "353",
      "reason": undefined,
      "support": 0,
      "votesMantissa": "1e+23",
    },
  ],
  "againstVotesMantissa": "1e+23",
  "cancelDate": undefined,
  "cancelTxHash": undefined,
  "createdDate": 2023-09-20T07:39:35.000Z,
  "createdTxHash": "0xbe4f9879425120453393641d4518e571b289d3ec74e3a27a2d4281caf251f408",
  "description": {
    "abstainDescription": "I am indifferent to whether Venus Protocol proceeds or not",
    "againstDescription": "I do not think that Venus Protocol should proceed with this proposal",
    "description": "This VIP will perform the following Risk Parameter actions as per Chaos Labsâ€™ latest recommendations in this Venus community forum publication: [Chaos Labs - Risk Parameter Updates - 08/02/24](https://community.venus.io/t/chaos-labs-risk-parameter-updates-08-02-24/4502).

- [WETH (Liquid Staked ETH)](https://etherscan.io/address/0xc82780Db1257C788F262FBbDA960B3706Dfdcaf2):
    - Reduce Collateral Factor, from 90% to 0%
    - Increase Kink, from 80% to 90%
    - Increase Multiplier, from 3.5% to 4.5%
    - Increase Reserve Factor, from 15% to 25%

Complete analysis and details of these recommendations are available in the above publication.

Please note that reducing the collateral factor to 0% will not cause current borrowers to be liquidated as the liquidation threshold will not be modified.

#### References

- [VIP simulation](https://github.com/VenusProtocol/vips/pull/356)

#### Disclaimer for Ethereum VIPs

Privilege commands on Ethereum will be executed by the [Guardian wallet](https://etherscan.io/address/0x285960C5B22fD66A736C7136967A3eB15e93CC67), until the [Multichain Governance](https://docs-v4.venus.io/technical-reference/reference-technical-articles/multichain-governance) contracts are fully enabled. If this VIP passes, [this](https://app.safe.global/transactions/tx?safe=eth:0x285960C5B22fD66A736C7136967A3eB15e93CC67&id=multisig_0x285960C5B22fD66A736C7136967A3eB15e93CC67_0xab824acf17443e1c50231fa0d6ad1d854269941db67704518104867ac9181083) multisig transaction will be executed. Otherwise, it will be rejected.",
    "forDescription": "I agree that Venus Protocol should proceed with this proposal",
    "title": "VIP-353 [Ethereum] Risk Parameters Adjustments (WETH)",
    "version": "v2",
  },
  "endBlock": 33499859,
  "endDate": 2023-06-15T09:07:50.000Z,
  "executedDate": undefined,
  "executedTxHash": "0x91f6267525b678fe62632dc738c766ccb02408a81cd9f5915f4f40db79368ce2",
  "executionEtaDate": 1970-01-01T00:00:00.000Z,
  "forVotes": [
    {
      "address": "0xc444949e0054a23c44fc45789738bdf64aed2391",
      "proposalId": "353",
      "reason": "Community delegators approved.",
      "support": 1,
      "votesMantissa": "4.33469577024822530748692e+23",
    },
    {
      "address": "0xc72f3295b7fe847ec99e3a3c6dd6e90db5e1061b",
      "proposalId": "353",
      "reason": undefined,
      "support": 1,
      "votesMantissa": "1.00001059438691098785833e+23",
    },
    {
      "address": "0xfce889ba7afde7428acaa3ef146b1bd526e78a19",
      "proposalId": "353",
      "reason": undefined,
      "support": 1,
      "votesMantissa": "1e+23",
    },
    {
      "address": "0x09c90ea0d16ce058229b939967aa29446d13ceed",
      "proposalId": "353",
      "reason": undefined,
      "support": 1,
      "votesMantissa": "1.16937e+21",
    },
    {
      "address": "0x53beb1e70d3d051d1e8b061ab93a6212c728317d",
      "proposalId": "353",
      "reason": undefined,
      "support": 1,
      "votesMantissa": "199226357358097054944",
    },
    {
      "address": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
      "proposalId": "353",
      "reason": undefined,
      "support": 1,
      "votesMantissa": "13777102348559360929",
    },
    {
      "address": "0x51e76fd862d563d58b7a807b01eabe0f3b07ac1a",
      "proposalId": "353",
      "reason": undefined,
      "support": 1,
      "votesMantissa": "13058470954295420970",
    },
    {
      "address": "0x627c97d755fc7fbb54ceb3c8e7d341cf9074c357",
      "proposalId": "353",
      "reason": undefined,
      "support": 1,
      "votesMantissa": "9977051682906883578",
    },
  ],
  "forVotesMantissa": "6.34876045445857488254946e+23",
  "proposalActions": [],
  "proposalId": "353",
  "proposalType": 1,
  "proposerAddress": "0xc444949e0054a23c44fc45789738bdf64aed2391",
  "queuedDate": undefined,
  "queuedTxHash": "0xc295f4f351eb2ebeebbbee99066733585458ef91fc38570594c8da1e0554cfd6",
  "startDate": 2023-06-15T09:00:20.000Z,
  "state": 4,
  "totalVotesMantissa": "7.3567789715995909474803e+23",
  "userVoteSupport": 1,
}
`;
